hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDiffUtils.java

```diff
@@ -101,8 +101,11 @@ public static boolean doesSstFileContainKeyRange(String filepath,
             ManagedSstFileReaderIterator iterator = ManagedSstFileReaderIterator.managed(
                 sstFileReader.newIterator(readOptions))) {
           iterator.get().seek(prefix.getBytes(UTF_8));
-          String seekResultKey = new String(iterator.get().key(), UTF_8);
-          return seekResultKey.startsWith(prefix);
+          if (iterator.get().isValid()) {
```

Comment:
Let's check the prefix against the first and last keys using [isKeyWithPrefixPresent](https://github.com/apache/ozone/blob/master/hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDiffUtils.java#L49) e.g. firstKey <= prefix <= lastKey.